File: android-app/app/src/main/java/com/restoloc/app/data/Restaurant.kt
package com.restoloc.app.data

import com.google.gson.annotations.SerializedName

data class Restaurant(
    @SerializedName("id") val id: Int? = null,
    @SerializedName("nom") val nom: String? = "",
    @SerializedName("telephone") val telephone: String? = "",
    @SerializedName("origine_poulet") val originePoulet: String? = "",
    @SerializedName("origine_viande") val origineViande: String? = "",
    @SerializedName("latitude") val latitude: Double? = 0.0,
    @SerializedName("longitude") val longitude: Double? = 0.0,
    @SerializedName("type") val type: String? = "Physique",
    @SerializedName("est_fiable") val estFiable: Boolean? = true,
    @SerializedName("lien_commande") val lienCommande: String? = "",
    @SerializedName("restaurant_type") val restaurantType: String? = "",
    @SerializedName("food") val food: String? = ""
)

File: Restaurant.cs
namespace RestoLoc;

using System.Text.Json.Serialization;

public enum TypeRestaurant { Physique, EnLigne }

public class Restaurant
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("nom")]
    public string Nom { get; set; } = string.Empty;

    [JsonPropertyName("telephone")]
    public string Telephone { get; set; } = string.Empty;

    [JsonPropertyName("origine_poulet")]
    public string OriginePoulet { get; set; } = string.Empty;

    [JsonPropertyName("origine_viande")]
    public string OrigineViande { get; set; } = string.Empty;

    [JsonPropertyName("latitude")]
    public double Latitude { get; set; }

    [JsonPropertyName("longitude")]
    public double Longitude { get; set; }
    
    [JsonPropertyName("type")]
    public TypeRestaurant Type { get; set; } = TypeRestaurant.Physique;

    [JsonPropertyName("est_fiable")]
    public bool EstFiable { get; set; } = true;

    [JsonPropertyName("lien_commande")]
    public string LienCommande { get; set; } = string.Empty;

    [JsonPropertyName("restaurant_type")]
    public string RestaurantType { get; set; } = string.Empty;

    [JsonPropertyName("food")]
    public string Food { get; set; } = string.Empty;

    [JsonPropertyName("created_at")]
    public DateTime? CreatedAt { get; set; }
}